editor/src/components/canvas/canvas-strategies/draw-to-insert-strategy.tsx

```diff
@@ -0,0 +1,329 @@
+    ) {
+      const insertionSubject = canvasState.interactionTarget.subjects[0]
+
+      const insertionCommand = getInsertionCommands(insertionSubject, interactionState, null)
+
+      if (insertionCommand != null) {
+        const reparentCommand = updateFunctionCommand(
+          'always',
+          (editorState): Array<EditorStatePatch> => {
+            return runTargetStrategiesForFreshlyInsertedElementToReparent(
+              canvasState.builtInDependencies,
+              editorState,
+              strategyState,
+              interactionState,
+              insertionSubject,
+              insertionCommand.frame,
+            )
+          },
+        )
+
+        const resizeCommand = updateFunctionCommand(
+          'always',
+          (editorState, commandLifecycle): Array<EditorStatePatch> => {
+            return runTargetStrategiesForFreshlyInsertedElementToResize(
+              canvasState.builtInDependencies,
+              editorState,
+              strategyState,
+              interactionState,
+              commandLifecycle,
+              insertionSubject,
+              insertionCommand.frame,
+            )
+          },
+        )
+
+        return strategyApplicationResult([insertionCommand.command, reparentCommand, resizeCommand])
+      }
+    }
+    // Fallback for when the checks above are not satisfied.
+    return emptyStrategyApplicationResult
+  },
+}
+
+function getInsertionCommands(
+  subject: ElementInsertionSubject,
+  interactionState: InteractionSession,
+  parent: ElementPath | null,
+): { command: InsertElementInsertionSubject; frame: CanvasRectangle } | null {
+  if (
+    interactionState.interactionData.type === 'DRAG' &&
+    interactionState.interactionData.drag != null
+  ) {
+    const pointOnCanvas = interactionState.interactionData.dragStart
+
+    const frame = canvasRectangle({
+      x: pointOnCanvas.x,
+      y: pointOnCanvas.y,
+      width: 0,
+      height: 0,
+    })
+
+    const updatedAttributesWithPosition = getStyleAttributesForFrameInAbsolutePosition(
+      subject,
+      frame,
+    )
+
+    const updatedInsertionSubject: ElementInsertionSubject = {
+      ...subject,
+      parent: parent
+        ? { target: parent, staticTarget: EP.dynamicPathToStaticPath(parent) }
+        : subject.parent,
+      element: {
+        ...subject.element,
+        props: updatedAttributesWithPosition,
+      },
+    }
+
+    return {
+      command: insertElementInsertionSubject('always', updatedInsertionSubject),
+      frame: frame,
+    }
+  }
+  return null
+}
+
+function getStyleAttributesForFrameInAbsolutePosition(
+  subject: ElementInsertionSubject,
+  frame: CanvasRectangle,
+) {
+  const updatedAttributes = LayoutHelpers.updateLayoutPropsWithFrame(
+    false,
+    null,
+    subject.element.props,
+    {
+      left: frame.x,
+      top: frame.y,
+      width: frame.width,
+      height: frame.height,
+    },
+    ['style'],
+  )
+
+  if (isLeft(updatedAttributes)) {
```

Comment:
Maybe a bit cleaner with a `foldEither`.

Reply:
Fixed in d22fe17f8bdd0ad0bb7d8bc117885150d0f44a25